pkg/controller/sparkapplication/controller.go

```diff
@@ -494,6 +494,9 @@ func (c *Controller) syncSparkApplication(key string) error {
 			appToUpdate = c.submitSparkApplication(appToUpdate)
 		}
 	case v1beta1.InvalidatingState:
+		if err := c.getAndUpdateAppState(appToUpdate); err != nil {
```

Comment:
Put more thoughts into this. Instead of doing this here, which does a lot more things than getting the driver pod name, which is what is essentially needed, how about add logic in `deleteSparkResources` to first list and find the driver pod, and then does the deletion? We can abstract the code for listing and finding the driver pod into a utility method and use it in both `deleteSparkResources` and `getAndUpdateAppState`.

Reply:
Sounds good to me. I'll make the changes and push them later today.

Reply:
I refactored the code a little bit so that the original `getAndUpdateAppState` is split into two sub-functions: `getAndUpdateDriverState` and `getAndUpdateExecutorState`. The former is what's needed in `deleteSparkResources`.